src/coinbase/address/developer_address.ts

```diff
@@ -0,0 +1,364 @@
+   * @param toAssetId - The ID of the Asset to trade to.
+   * @returns The Trade object.
+   * @throws {Error} If the private key is not loaded, or if the asset IDs are unsupported, or if there are insufficient funds.
+   */
+  public async createTrade(amount: Amount, fromAssetId: string, toAssetId: string): Promise<Trade> {
+    const fromAsset = await Asset.fetch(this.getNetworkId(), fromAssetId);
+    const toAsset = await Asset.fetch(this.getNetworkId(), toAssetId);
+    const wallet = new ethers.Wallet(this._key!.privateKey);
+
+    await this.validateCanTrade(amount, fromAssetId);
+    const trade = await this.createTradeRequest(amount, fromAsset, toAsset);
+    // NOTE: Trading does not yet support server signers at this point.
+    const signed_payload = await trade.getTransaction().sign(wallet);
+    const approveTransactionSignedPayload = trade.getApproveTransaction()
+      ? await trade.getApproveTransaction()!.sign(wallet)
+      : undefined;
+
+    return this.broadcastTradeRequest(trade, signed_payload, approveTransactionSignedPayload);
+  }
+
+  /**
+   * Creates a trade model for the specified amount and assets.
+   *
+   * @param amount - The amount of the Asset to send.
+   * @param fromAsset - The Asset to trade from.
+   * @param toAsset - The Asset to trade to.
+   * @returns A promise that resolves to a Trade object representing the new trade.
+   */
+  private async createTradeRequest(
+    amount: Amount,
+    fromAsset: Asset,
+    toAsset: Asset,
+  ): Promise<Trade> {
+    const tradeRequestPayload = {
+      amount: fromAsset.toAtomicAmount(new Decimal(amount.toString())).toString(),
+      from_asset_id: fromAsset.primaryDenomination(),
+      to_asset_id: toAsset.primaryDenomination(),
+    };
+    const tradeModel = await Coinbase.apiClients.trade!.createTrade(
+      this.getWalletId(),
+      this.getId(),
+      tradeRequestPayload,
+    );
+    return new Trade(tradeModel?.data);
+  }
+
+  /**
+   * Broadcasts a trade using the provided signed payloads.
+   *
+   * @param trade - The Trade object representing the trade.
+   * @param signedPayload - The signed payload of the trade.
+   * @param approveTransactionPayload - The signed payload of the approval transaction, if any.
+   * @returns A promise that resolves to a Trade object representing the broadcasted trade.
+   */
+  private async broadcastTradeRequest(
+    trade: Trade,
+    signedPayload: string,
+    approveTransactionPayload?: string,
+  ): Promise<Trade> {
+    const broadcastTradeRequestPayload = {
+      signed_payload: signedPayload,
+      approve_transaction_signed_payload: approveTransactionPayload
+        ? approveTransactionPayload
+        : undefined,
+    };
+
+    const response = await Coinbase.apiClients.trade!.broadcastTrade(
+      this.getWalletId(),
+      this.getId(),
+      trade.getId(),
+      broadcastTradeRequestPayload,
+    );
+
+    return new Trade(response.data);
+  }
+
```

Comment:
We also need `export` method

Reply:
Can you clarify what method to export? All I'm seeing in the tagged comment is a return on the `Trade` object.

Reply:
sorry, I mean we have `export` method on WalletAddress: https://github.com/coinbase/coinbase-sdk-ruby/blob/v0.0.9/lib/coinbase/address/wallet_address.rb#L125

Reply:
Chatted offline, not needed.